lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -67,10 +59,59 @@ using namespace PlatformAgnostic::ICUHelpers;
         ICU_ASSERT(__status, __len > 0 && __len < localeIDLen);               \
     } while (false)
 
+#define ASSERT_ENUM(T, e)                                                     \
+    do                                                                        \
+    {                                                                         \
+        if ((int)(e) < 0 || (e) >= T::Max)                                    \
+        {                                                                     \
+            AssertMsg(false, #e " of type " #T " has an invalid value");      \
+        }                                                                     \
+    } while (false)
+
 #endif // INTL_ICU
 
-#include "PlatformAgnostic/Intl.h"
-using namespace PlatformAgnostic::Intl;
+// NOTE(jahorto): Keep these enums in sync with those by the same name in Intl.js
+// These enums are used by both WinGlob- and ICU-backed Intl
+enum class NumberFormatStyle
+{
+    Decimal, // Intl.NumberFormat(locale, { style: "decimal" }); // aka in our code as "number"
+    Percent, // Intl.NumberFormat(locale, { style: "percent" });
+    Currency, // Intl.NumberFormat(locale, { style: "currency", ... });
+
+    Max,
+    Default = Decimal,
+};
+
+enum class NumberFormatCurrencyDisplay
+{
+    Symbol, // Intl.NumberFormat(locale, { style: "currency", currencyDisplay: "symbol" }); // e.g. "$" or "US$" depeding on locale
+    Code, // Intl.NumberFormat(locale, { style: "currency", currencyDisplay: "code" }); // e.g. "USD"
+    Name, // Intl.NumberFormat(locale, { style: "currency", currencyDisplay: "name" }); // e.g. "US dollar"
+
+    Max,
+    Default = Symbol,
+};
+
+enum class CollatorSensitivity
+{
+    Base,
+    Accent,
+    Case,
+    Variant,
+
+    Max,
+    Default = Variant,
+};
+
+enum class CollatorCaseFirst
+{
+    Upper,
+    Lower,
+    False,
+
+    Max,
+    Default = False,
```

Comment:
Side note: feels a bit weird for a false value to be non-zero, but I see that's how it was before as well

Reply:
IIRC it's basically spec names and defaults. Not entirely sure whether there's anything forcing the default value not to be 0 (like numbers formally assigned to these meanings by some API).

---
In reply to: [171408485](https://github.com/Microsoft/ChakraCore/pull/4743#discussion_r171408485) [](ancestors = 171408485)